### ISSUE TITLE
 Found errors in Test_python3_chdir()

### ISSUE THREAD

**Issue by riesebie** [vim/vim#6405]
Compiling 8.2.1145 within a tmux session using shadow dirs I get:
``` Failures:                                                                                                                                                                                                     
          From test_python3.vim:                                                                                                                                                                                
          Found errors in Test_python3_chdir():                                                                                                                                                                 
          Caught exception in Test_python3_chdir(): Vim(py3):NameError: name 'path' is not defined @ function RunTheTest[39]..Test_python3_chdir, line 27 
```

**Comment by vim-ml** [comment]
Hi,

On Mon, Jul 6, 2020 at 3:35 AM Elimar Riesebieter <vim-dev-github@256bit.org>
wrote:

> Compiling 8.2.1145 within a tmux session using shadow dirs I get:
>
>           From test_python3.vim:
>           Found errors in Test_python3_chdir():
>           Caught exception in Test_python3_chdir(): Vim(py3):NameError: name 'path' is not defined @ function RunTheTest[39]..Test_python3_chdir, line 27
>
>
>
Can you try applying the following change to that test?

https://github.com/vim/vim/commit/d3cbebbb558d0b49bc7cbe1d304d355fb709e35a

Thanks,
Yegappan

**Comment by chrisbra** [comment]
Hm, test_python3_chdir() is not that long:

https://github.com/vim/vim/blob/0ad3e894d75236915e67dfbbcc821b6bb3c05d91/src/testdir/test_python3.vim#L2294-L2325

Hm, I see there is a `del path` twice, one in line 2312 and 2316. Perhaps that is the problem?
It also contains a comment about running in a shadow dir. Can you check it and see where the problem lives? Might also help to step through the test manually to see where it fails.

Thanks.

**Comment by riesebie** [comment]
> Can you try applying the following change to that test? [d3cbebb](https://github.com/vim/vim/commit/d3cbebbb558d0b49bc7cbe1d304d355fb709e35a) Thanks, Yegappan

Yeah, that fixed it but with a slightly different patch (different line numbers):
```
diff --git a/src/testdir/test_python3.vim b/src/testdir/test_python3.vim  
  index 7edf7d055..201b2eafe 100644                                                                                                                                                                             
  --- a/src/testdir/test_python3.vim                                                                                                                                                                            
  +++ b/src/testdir/test_python3.vim                                                                                                                                                                            
  @@ -2309,7 +2309,6 @@ func Test_python3_chdir()                                                                                                                                                               
         cb.append(str(fnamemodify('.', ':p:h:t')))                                                                                                                                                             
         cb.append(vim.eval('@%')[len(path)+1:].replace(os.path.sep, '/'))                                                                                                                                      
         os.chdir(path)                                                                                                                                                                                         
  -      del path                                                                                                                                                                                               
       else:                                                                                                                                                                                                    
         cb.append(str(fnamemodify('.', ':p:h:t')))                                                                                                                                                             
         cb.append(vim.eval('@%').replace(os.path.sep, '/') 
```

**Comment by riesebie** [comment]
* Elimar Riesebieter <notifications@github.com> [2020-07-07 00:32 -0700]:

> > Can you try applying the following change to that test? [d3cbebb](https://github.com/vim/vim/commit/d3cbebbb558d0b49bc7cbe1d304d355fb709e35a) Thanks, Yegappan
> 
> Yeah, that fixed it but with a slightly different patch (different line numbers):

Seems to be fixed with 8.2.1146.

Thanks
Elimar
-- 
  Learned men are the cisterns of knowledge,
  not the fountainheads ;-)
